streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -418,7 +417,7 @@ void closeSuspended(boolean clean, RuntimeException firstException) {
                     try {
                         producer.abortTransaction();
                         transactionInFlight = false;
-                    } catch (final ProducerFencedException e) {
+                    } catch (final KafkaException e) {
```

Comment:
If a producer gets fenced (for example due to GC pause) and we close it as zombie, the call to `producer.abortTransaction` does throw a `KafkaException` but not `ProducerFenced` exception. If we only catch `ProducerFenced` the rest of the method would not be executed and thus, we don't close the producer and it's client-id is never deregistered from the metrics. This leads to ID conflict when the new tasks starts as the new task taking over will use the same client-id as the fenced producer (this issue was detected by one of the EOS integration tests).

The only alternative here would be to add a flag (for regular close vs zombie close) and don't call `abortTransaction` for the zombie case. But this seems to be unnecessary complex.